### ISSUE TITLE
ValueError: Got unexpected message type: AIMessageChunk

### ISSUE THREAD

**Issue by 1nnovat1on** [AntonOsika/gpt-engineer#804]
## Policy and info
 - Maintainers will close issues that have been stale for 14 days if they contain relevant answers.
 - Adding the label "sweep" will automatically turn the issue into a coded pull request. Works best for mechanical tasks. More info/syntax at: https://docs.sweep.dev/

## Expected Behavior

I typed in a prompt after setting it up. When it responds with clarifying questions, I expect to be able to respond.

## Current Behavior

It fails with the error message in the title.

## Failure Information

Windows, GPT-4, CMD, 

### Steps to Reproduce

If possible, provide detailed steps for reproducing the issue.

1. pip install gpt-engineer
2. Created folder
3. enter gpt-engineer "C:\python\test_project" "gpt-4"
4. Get back some feedback, respond to a few
5. then error

### Failure Logs

│ s.py:336 in _message_from_dict                                                                   │
│                                                                                                  │
│   333 │   elif _type == "function":                                                              │
│   334 │   │   return FunctionMessage(**message["data"])                                          │
│   335 │   else:                                                                                  │
│ ❱ 336 │   │   raise ValueError(f"Got unexpected message type: {_type}")                          │
│   337                                                                                            │
│   338                                                                                            │
│   339 def messages_from_dict(messages: List[dict]) -> List[BaseMessage]:                         │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │   _type = 'AIMessageChunk'                                                                   │ │
│ │ message = {                                                                                  │ │
│ │           │   'type': 'AIMessageChunk',                                                      │ │
│ │           │   'data': {                                                                      │ │
│ │           │   │   'content': 'Summary of areas that need clarification:\n\n1. The specific   │ │
│ │           type of reinforcement'+377,                                                        │ │
│ │           │   │   'additional_kwargs': {},                                                   │ │
│ │           │   │   'type': 'AIMessageChunk',                                                  │ │
│ │           │   │   'example': False                                                           │ │
│ │           │   }                                                                              │ │
│ │           }                                                                                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: Got unexpected message type: AIMessageChunk

**Comment by wwencel** [comment]
```
Traceback (most recent call last):

  File "/Users/wlodek/dev/venv-gpt-eng/bin/gpt-engineer", line 8, in <module>
    sys.exit(app())

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/gpt_engineer/main.py", line 96, in main
    messages = step(ai, dbs)

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/gpt_engineer/steps.py", line 192, in gen_clarified_code
    messages = AI.deserialize_messages(dbs.logs[clarify.__name__])

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/gpt_engineer/ai.py", line 216, in deserialize_messages
    return list(messages_from_dict(json.loads(jsondictstr)))  # type: ignore

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/langchain/schema/messages.py", line 348, in messages_from_dict
    return [_message_from_dict(m) for m in messages]

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/langchain/schema/messages.py", line 348, in <listcomp>
    return [_message_from_dict(m) for m in messages]

  File "/Users/wlodek/dev/venv-gpt-eng/lib/python3.10/site-packages/langchain/schema/messages.py", line 336, in _message_from_dict
    raise ValueError(f"Got unexpected message type: {_type}")

ValueError: Got unexpected message type: AIMessageChunk
```
same error I got, but it looks like langchain issue

**Comment by Fw-JP** [comment]
`Traceback (most recent call last):

  File "<frozen runpy>", line 198, in _run_module_as_main

  File "<frozen runpy>", line 88, in _run_code

  File "E:\VSC Workspaces\venv\Scripts\gpt-engineer.exe\__main__.py", line 7, in <module>
    sys.exit(app())
             ^^^^^

               ^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\gpt_engineer\steps.py", line 192, in gen_clarified_code
    messages = AI.deserialize_messages(dbs.logs[clarify.__name__])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\gpt_engineer\ai.py", line 216, in deserialize_messages
    return list(messages_from_dict(json.loads(jsondictstr)))  # type: ignore
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 348, in messages_from_dict
    return [_message_from_dict(m) for m in messages]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 348, in <listcomp>
    return [_message_from_dict(m) for m in messages]
            ^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 336, in _message_from_dict
    raise ValueError(f"Got unexpected message type: {_type}")

ValueError: Got unexpected message type: AIMessageChunk`

Similar issue here. Anything I can do?

**Comment by Fw-JP** [comment]
```
Traceback (most recent call last):

  File "<frozen runpy>", line 198, in _run_module_as_main

  File "<frozen runpy>", line 88, in _run_code

  File "E:\VSC Workspaces\venv\Scripts\gpt-engineer.exe\__main__.py", line 7, in <module>
    sys.exit(app())
             ^^^^^

               ^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\gpt_engineer\steps.py", line 192, in gen_clarified_code
    messages = AI.deserialize_messages(dbs.logs[clarify.__name__])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\gpt_engineer\ai.py", line 216, in deserialize_messages
    return list(messages_from_dict(json.loads(jsondictstr)))  # type: ignore
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 348, in messages_from_dict
    return [_message_from_dict(m) for m in messages]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 348, in <listcomp>
    return [_message_from_dict(m) for m in messages]
            ^^^^^^^^^^^^^^^^^^^^^

  File "E:\VSC Workspaces\venv\Lib\site-packages\langchain\schema\messages.py", line 336, in _message_from_dict
    raise ValueError(f"Got unexpected message type: {_type}")

ValueError: Got unexpected message type: AIMessageChunk
```

Same here. Anything I can do ?

**Comment by paulternate** [comment]
I traced the error back a little bit to a missing handler in `C:\Python\lib\site-packages\langchain\schema\messages.py`  and was able to get past it by adding this to the `_message_from_dict` function right before the else statement. 

```python
elif _type == "AIMessageChunk":
        return AIMessageChunk(**message["data"])
```

**Comment by pralad-p** [comment]
I think how `gpt-engineer` deals with serialization of user-entered messages (because the OP says it didn't happen on the prompt file input, but rather on the subsequent messages) could indicate that the implicit properties added during serialization (or this AIMessageChunk type cast) has a strong dependency to how Langchain modifies its schema as per its requirements.  What I mean is previous Langchain might have been quite loose with the type of messages it recieved, but now (I assume due to extra features) needed the type separation. Maybe it's a good idea to brainstorm ways on how to loosen the coupling of this dependency. Additional tests in this dept. would be the first step to know the scope of the problem.

**Comment by ibrahim-sowunmi** [comment]
> ```python
> elif _type == "AIMessageChunk":
>         return AIMessageChunk(**message["data"])
> ```

Fixed it for me on mac.

`python3 -c "import site; print(site.getsitepackages())"
`
Then open the schema with open . and edited the file

**Comment by ATheorell** [comment]
@wwencel @0x1pikachu 
Can you rerun on the latest main commit or at tag 0.1.0?

The stack trace 
  File "E:\VSC Workspaces\venv\Lib\site-packages\gpt_engineer\ai.py", line 216, in deserialize_messages
    return list(messages_from_dict(json.loads(jsondictstr)))  # type: ignore
  is not up to date.

**Comment by ATheorell** [comment]
@pralad-p agreed. 
I think the first thing that needs to be done is to switch to better dependency management so we get less unpleasant surprises when some dependency is updated.

**Comment by MWals** [comment]
> > ```
> > elif _type == "AIMessageChunk":
> >         return AIMessageChunk(**message["data"])
> > ```
> 
> Fixed it for me on mac.
> 
> `python3 -c "import site; print(site.getsitepackages())" ` Then open the schema with open . and edited the file

This also fixed the problem for me on Mac. Please consider integrating this into the code.

**Comment by ATheorell** [comment]
@MWals @ibrahim-sowunmi @paulternate 
The code you are referring to is inside langchain and not gpt-engineer. Thus, we cannot modify it. It would be awesome if you could request this change at https://github.com/langchain-ai/langchain

**Comment by ATheorell** [comment]
A direct fix here, instead of modifying langchain, is probably to downgrade the langchain version in your environment.
So far, nobody has provided an example that I can reproduce, which would help me in debugging this.

**Comment by TheoMcCabe** [comment]
ok So i also, from reading this issue, thought these guys were talking about editing code in langchain. This was confusing 

"I traced the error back a little bit to a missing handler in C:\Python\lib\site-packages\langchain\schema\messages.py and was able to get past it by adding this to the _message_from_dict function right before the else statement."

However they are all actually suggesting a fix to the gpt-engineer code for this. 

Tbh something like - it's better to raise a PR than get bogged down in comments on an issue. 

Thanks for the suggestion @paulternate  i will raise a PR for this change now

**Comment by ATheorell** [comment]
This is now addressed in PR #833 by restricting the langchain version. Reinstalling from main branch source should solve it.

**Comment by jonnolen** [comment]
the update from this issue landed in upstream langchain really quickly... https://github.com/langchain-ai/langchain/pull/15327.  credit to @paulternate
